### DISCUSSION TITLE
UI Mockup for discussion

### DISCUSSION THREAD

**Discussion by davepl** [PlummersSoftwareLLC/NightDriverStrip#559] [Announcements]
The UI has made great leaps in the last few months, and I wanted to set a bit of a direction for the look so that its usable on both the phone and the desktop.   Basically, I'm arguing we should target the phone, then make extra use of the desktop space if and when it's useful.

To that end, I mocked up the essentials of the front page and the status page.  It doesn't show everything, like settings, as they would follow the same general style.

Please let me know your thoughts and concerns!

![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/3698967/8d3db5bf-421d-4d12-9d07-13d4e229abb3)

![image](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/3698967/9535fbb4-aeff-4b4b-9184-250e1b36ac38)

**Comment by robertlipe** [comment]
Nice.

I hope mostly this can be done with media types in css or similar
techniques and comparatively little code.

**Comment by revision29** [comment]
That looks nice. The only feedback so far is that the play button also looks like a right navigation / disclosure triangle. In context I know what it is, but years of iOS use has conditioned me to think if I tap an effect, I will be brought to another view. 

Having the color scheme displayed is cool. How will that handle a 16 color palette? Will a person be able to tweak the effect's variables or even change the colors it is using (if the effect supports that)?

**Reply by rbergen** [reply]
> Having the color scheme displayed is cool. How will that handle a 16 color palette? Will a person be able to tweak the effect's variables or even change the colors it is using (if the effect supports that)?

The color scheme shown in this mockup consists of the two (primary and secondary) "global colors". Effects can decide how to apply those to what they show, including not applying them at all - although issue #485 aims to "generally dissuade" effects from using that latter approach.

On top of the global colors, each effect can use its own colors/palettes and choose to expose those as changeable settings. These colors/palettes would then show up on the settings page for the effect in question. That's a "per effect" choice, and as such requires work for each effect. This is covered by issue #294.
